ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,269 @@
+
+        return container
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton(type: .system)
+        return button
+    }()
+
+    private let badgeView: UILabel = {
+        let badgeView = UILabel(frame: .zero)
+        badgeView.layer.masksToBounds = true
+        badgeView.backgroundColor = Colors.Palette.dangerPrimary.color
+        badgeView.textColor = .white
+        badgeView.textAlignment = .center
+        badgeView.font = UIFont.systemFont(ofSize: Constants.badgeFontSize, weight: .regular)
+        badgeView.isHidden = true
+
+        return badgeView
+    }()
+
+    private func updateLayout() {
+        updateButton()
+        updateBadgeView()
+        invalidateIntrinsicContentSize()
+    }
+
+    private func updateButton() {
+        button.isEnabled = item.isEnabled
+
+        if isLeftItem {
+            let isRTL = effectiveUserInterfaceLayoutDirection == .rightToLeft
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: isRTL ? 0 : Constants.leftBarButtonItemLeadingMargin, bottom: 0, right: isRTL ? Constants.leftBarButtonItemLeadingMargin : 0)
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: Constants.rightBarButtonItemHorizontalPadding, bottom: 0, right: Constants.rightBarButtonItemHorizontalPadding)
+        }
+
+        button.tag = item.tag
+        button.tintColor = item.tintColor
+        button.titleLabel?.font = item.titleTextAttributes(for: .normal)?[.font] as? UIFont
+
+        var portraitImage = item.image
+        if portraitImage?.renderingMode == .automatic {
+            portraitImage = portraitImage?.withRenderingMode(.alwaysTemplate)
+        }
+        var landscapeImage = item.landscapeImagePhone ?? portraitImage
+        if landscapeImage?.renderingMode == .automatic {
+            landscapeImage = landscapeImage?.withRenderingMode(.alwaysTemplate)
+        }
+
+        button.setImage(traitCollection.verticalSizeClass == .regular ? portraitImage : landscapeImage, for: .normal)
+        button.setTitle(item.title, for: .normal)
+
+        if let action = item.action {
+            button.addTarget(item.target, action: action, for: .touchUpInside)
+        }
+
+        button.showsLargeContentViewer = true
+
+        if let customLargeContentSizeImage = item.largeContentSizeImage {
+            button.largeContentImage = customLargeContentSizeImage
+        }
+
+        if item.title == nil {
+            button.largeContentTitle = item.accessibilityLabel
+        }
+
+        if #available(iOS 13.4, *) {
+            // Workaround check for beta iOS versions missing the Pointer Interactions API
+            if arePointerInteractionAPIsAvailable() {
+                button.isPointerInteractionEnabled = true
+            }
+        }
+    }
+
+    private func updateBadgeView() {
+        badgeView.text = badgeValue
+        badgeView.isHidden = badgeValue == nil
+
+        if badgeValue != nil {
```

Comment:
please use guard-let to avoid a lot of optional chaining operations in the below code-base.

Reply:
Actually, in this case if else makes more sense because:
1. We are not going to use badgeValue inside if block.
2. There is a line of code in else block and so it's not just return statement in else block of guard.